### PR TITLE
Handles embedded object fields and lists (V1 only)

### PR DESCRIPTION
**Pull request by ypc-faros**
## Description

Handles embedded object fields and lists

## Type of change
- [X] Bug fix
- [ ] New feature
- [ ] Breaking change

## Related issues

> Fix [#1]() 

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

File: src/graphql/graphql.ts [hunk 1]
```diff
@@ -1045,6 +1060,19 @@ function isV1ModelType(type: any): type is gql.GraphQLObjectType {
   );
 }
 
+function isV1EmbeddedType(type: any): type is gql.GraphQLObjectType {
+  return (
+    gql.isObjectType(type) &&
+    !some(type.getInterfaces(), (i) => i.name === 'Node') &&
+    type.name !== '__Schema' &&
```
Comment (ypc-faros) [inline src/graphql/graphql.ts:1067]: This is sad, but it works for our schema.

It was getting an infinite recursion on the embedded resolver until I filtered these out. If you have any ideas on how to improve, I'm all ears.

File: test/resources/v1_schema_for_embedded_test.gql [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+"""Generic node metadata"""
+type NodeMetadata {
+  """The origin of the entry which created the node"""
+  origin: String
+
+  """The latest timestamp of the entries affecting this node"""
+  refreshedAt: String
+}
+
+type Person {
```
Comment (ypc-faros) [inline test/resources/v1_schema_for_embedded_test.gql:123]: Here's the more relevant part to the introduced change

File: src/graphql/graphql.ts (hunk 1 above)
Comment (tovbinm) [reply src/graphql/graphql.ts:1067]: @eskrm @ted-faros ^

File: src/graphql/graphql.ts [hunk 3]
```diff
@@ -1793,3 +1826,52 @@ class PrimaryKeyResolver {
...
+    return result;
+  }
+
+  @Memoize((type: gql.GraphQLObjectType) => type.name)
+  private resolveEmbeddedField(type: gql.GraphQLObjectType): string {
+    const resolved = [];
+
+    for (const fldName of Object.keys(type.getFields())) {
+      const field = type.getFields()[fldName];
+      let unwrappedType = unwrapType(field.type) || field.type;
```
Comment (eskrm) [inline src/graphql/graphql.ts:1860]: Have `unwrapType` return the original type if it's already unwrapped.

File: src/graphql/graphql.ts [hunk 4]
```diff
@@ -1793,3 +1826,52 @@ class PrimaryKeyResolver {
...
+
+      if (gql.isListType(unwrappedType)) {
+        unwrappedType = unwrappedType.ofType;
```
Comment (eskrm) [inline src/graphql/graphql.ts:1863]: Also make it unwrap list types?

**Review by eskrm — APPROVED** [review]
LGTM

**Comment by ypc-faros** [comment]
Discussed offline with @eskrm 
I'll address some of this feedback in a separate PR.

File: src/graphql/graphql.ts (hunk 1 above)
Comment (tovbinm) [reply src/graphql/graphql.ts]: Thanks @eskrm !!

**Pull request merged**